infrastructure/aws/index.sh

```diff
@@ -45,16 +77,16 @@ echo "Channel is $CHANNEL"
 export AWS_ROOT=$(realpath $MOZSEARCH_PATH/infrastructure/aws)
 VOLUME_ID=$(python $AWS_ROOT/attach-index-volume.py $CHANNEL $EC2_INSTANCE_ID)
 
-# The EBS volume will no longer be mounted at /dev/xvdf but instead at an
-# arbitrarily assigned nvme id.  However, since we only have a single EBS volume
-# and we dynamically attach it, we're pretty certain what the ID will be:
-EBS_NVME_DEV=nvme2n1
+# Since we know the volume id and it's exposed as the `SerialNumber` in the JSON
+# structure (see above), we can look that up here too.  Note that we need to
+# remove the/any dash from the volume id.
+JQ_QUERY=".Devices[] | select(.SerialNumber == \"${VOLUME_ID/-/}\") | .DevicePath"
 
 set +o pipefail   # The grep command below can return nonzero, so temporarily allow pipefail
 for (( i = 0; i < 3600; i++ ))
 do
-    COUNT=$(lsblk | grep $EBS_NVME_DEV | wc -l)
-    if [ $COUNT -eq 1 ]
+    EBS_NVME_DEV=$(sudo nvme list -o json | jq --raw-output "$JQ_QUERY")
+    if [[ $EBS_NVME_DEV ]]
```

Comment:
What ends up happening here in the failure case? I would assume `EBS_NVME_DEV` is empty, so you'd want to do `-n "$EBS_NVME_DEV"` in the condition. Or does just putting the variable alone work in bash?